File: generate-secrets.js
#!/usr/bin/env node

/**
 * Generate secure random secrets for production use
 * Run with: node generate-secrets.js
 */

import crypto from 'crypto';

function generateSecret(length = 64) {
  return crypto.randomBytes(length).toString('hex');
}

console.log('\n🔐 Generated Secure Secrets for Production\n');
console.log('Copy these to your environment variables:\n');
console.log('═'.repeat(70));
console.log('\nJWT_SECRET=' + generateSecret(64));
console.log('\nSESSION_SECRET=' + generateSecret(64));
console.log('\n═'.repeat(70));
console.log('\n⚠️  Important: Keep these secrets secure and never commit them to git!\n');
console.log('💡 Tip: Generate new secrets for each environment (dev, staging, prod)\n');

File: backend/generate_secrets.py
#!/usr/bin/env python
"""Generate a secure random secret for production use.

Run with: python generate_secrets.py
"""

import secrets


def generate_secret(num_bytes: int = 64) -> str:
    return secrets.token_hex(num_bytes)


if __name__ == "__main__":
    print("\n🔐 Generated Secure Secret for Production\n")
    print("Copy this to your environment variables:\n")
    print("═" * 70)
    print(f"\nJWT_SECRET={generate_secret(64)}")
    print("\n" + "═" * 70)
    print("\n⚠️  Important: Keep this secret secure and never commit it to git!\n")
    print("💡 Tip: Generate a new secret for each environment (dev, staging, prod)\n")
